### PR TITLE
vector: fix suboptimal sol_ptr_vector_del_element implementation v2

### PR DESCRIPTION
**Pull request by kaspersky**
Signed-off-by: Vladimir Cernov <cernov.vladimir@gmail.com>

### CODE REVIEW

**Comment by kaspersky** [comment]
#1730 

I have avoided copying elements in the same place without using an if statement: by having 2 for loops (first one finds first occurence, second one removes elements).

**Comment by edersondisouza** [comment]
+1

File: src/lib/datatypes/sol-vector.c [hunk 1]
```diff
@@ -377,18 +377,23 @@ sol_vector_del_element(struct sol_vector *v, const void *elem)
...
-            SOL_INT_CHECK(r, < 0, r);
-            removed++;
-        }
+    uint16_t i = 0, offset = 0;
+    void **v = pv->base.data;
+
+    while (i < pv->base.len && v[i] != elem) {
+        i++;
+        offset++;
+    }
```
Comment (vcgomes) [inline src/lib/datatypes/sol-vector.c]: I am going to suggest the tiniest of optimizations, feel totally free to ignore: do `offset = i;` after the `while` loop.

**Comment by vcgomes** [comment]
Looking good.

File: src/lib/datatypes/sol-vector.c (hunk 1 above)
Comment (kaspersky) [inline src/lib/datatypes/sol-vector.c]: @vcgomes , good observation, will do. Wonder how I didn't think of it myself.

**Comment by kaspersky** [comment]
#2019

**Pull request closed without merging**